File: rails/app/models/destination.rb
class Destination < ActiveRecord::Base
    belongs_to :country
    has_many :flights, class_name: "Flight", foreign_key: 'destination_id'
    has_many :flights, class_name: "Flight", foreign_key: 'origin_id'

    has_many :journeys, class_name: "Journey", foreign_key: 'destination_id'
    has_many :journeys, class_name: "Journey", foreign_key: 'origin_id'

=begin
    Define and override our as_json method in order to generate custom json
    This json will include the required attributes of the desination as well
    as the country
=end
    
    def as_json(options={})
        
        {   
            :id => self.id,
            :destination_code => self.destination_code,
            :airport => self.airport,
            :country_code_2 => self.country.country_code_2,
            :country_code_3 => self.country.country_code_3,
            :country_name => self.country.country_name,
            :alternate_name_1 => self.country.alternate_name_1
        }

    end 
        

end

File: ember/flightpubember/app/models/destination.js
import DS from 'ember-data';

var Destination =  DS.Model.extend({
	
	destinationCode: 	DS.attr('string'),
	airport: 			DS.attr('string'),
	countryCode2: 		DS.attr('string'),
	countryCode3: 		DS.attr('string'),
	countryName: 		DS.attr('string'),
	alternateName1: 	DS.attr('string'),

	setFields: function(fields) {

		this.setProperties({
			destinationCode: 	fields['destination_code'],
			airport: 			fields['airport'],
			countryCode2: 		fields['country']['country_code_2'],
			countryCode3: 		fields['country']['country_code_3'],
			countryName: 		fields['country']['country_name'],
			alternateName1: 	fields['country']['alternate_name_1']
		});

	}
  
});

export default Destination;